src/networking/streams.rs

```diff
@@ -0,0 +1,80 @@
+use quinn::{ReadExactError, crypto::Session, generic::RecvStream};
+use quinn::{generic::SendStream, WriteError};
+
+use crate::networking::packets::*;
+
+#[derive(Debug)]
+pub struct BoundedPlanetSendStream<T: Session> {
+    send: SendStream<T>,
+}
+
+#[derive(Debug)]
+pub enum SendError {
+    EncodeError(rmp_serde::encode::Error),
+    WriteError(WriteError),
+}
+
+impl<TSession: Session> BoundedPlanetSendStream<TSession> {
+    pub fn new(send: SendStream<TSession>) -> BoundedPlanetSendStream<TSession> {
+        BoundedPlanetSendStream { send }
+    }
+
+    pub async fn send_packet<'a>(&mut self, packet: &Packet) -> Result<(), SendError> {
+        // Encode packet into messagepack format
+        let bytes = rmp_serde::to_vec(&packet).map_err(|e| SendError::EncodeError(e))?;
+
+        // Prefix with length (4 bytes, network order)
+        let len_bytes = (bytes.len() as u32).to_be_bytes();
+        self.send
+            .write_all(&len_bytes)
+            .await
+            .map_err(|e| SendError::WriteError(e))?;
+
+        // Write data to socket
+        self.send
+            .write_all(&bytes)
+            .await
+            .map_err(|e| SendError::WriteError(e))?;
+
+        return Ok(());
+    }
+}
+
+#[derive(Debug)]
+pub struct BoundedPlanetRecvStream<T: Session> {
+    recv: RecvStream<T>,
+}
+
+#[derive(Debug)]
+pub enum RecvError {
+    DecodeError(rmp_serde::decode::Error),
+    ReadExactError(ReadExactError),
+}
+
+impl<TSession: Session> BoundedPlanetRecvStream<TSession> {
+    pub fn new(recv: RecvStream<TSession>) -> BoundedPlanetRecvStream<TSession> {
```

Comment:
```suggestion
    pub const fn new(recv: RecvStream<TSession>) -> BoundedPlanetRecvStream<TSession> {
```
`const` where possible is always nice

Reply:
Can't be `const`, `TSession` does not allow it:

> trait bounds other than `Sized` on const fn parameters are unstable
> see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
> add `#![feature(const_fn)]` to the crate attributes to enablerustc(E0723)